pkg/network/common/egressip.go

```diff
@@ -301,6 +329,13 @@ func (eit *EgressIPTracker) syncEgressIPs() {
 	for ns := range changedNamespaces {
 		eit.syncEgressNamespaceState(ns)
 	}
+
+	if eit.updateEgressCIDRs {
+		eit.updateEgressCIDRs = false
+		if eit.nodesWithCIDRs > 0 {
+			eit.watcher.UpdateEgressCIDRs()
+		}
+	}
```

Comment:
Don't this need to be:
```
    if eit.updateEgressCIDRs {
        eit.updateEgressCIDRs = false
        eit.watcher.UpdateEgressCIDRs()
    }
```
Use case:
- We got a node with egress CIDR, so nodesWithCIDRs = 1 and watcher.UpdateEgressCIDRs will update corresponding hs.EgressIPs
- Now egress CIDR got removed from the same node, so nodeWithCIDRs = 0 and in this case we do need to call watcher.UpdateEgressCIDRs to remove old hs.EgressIPs?

Reply:
I debated about this. The problem is that `EgressCIDRs == nil` is defined to mean "don't automatically set EgressIPs on this node", so that behavior would be inconsistent with the normal behavior. (And it would cause problems if the controller restarted while the apiserver was still running, in which case the non-nil-to-nil transition might happen while the controller wasn't running, and then afterward it would have no idea that it was supposed to remove those EgressIPs.) And anyway, it's easy enough for the user to just unset EgressIPs themselves when they unset EgressCIDRs if that's the behavior they wanted.

I agree it's unexpected though. Maybe there should have been another field, `DoAutomaticEgressIPAssignment bool`, rather than overloading that into `EgressCIDRS == nil` / `!= nil`...